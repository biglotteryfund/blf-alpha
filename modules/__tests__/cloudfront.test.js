/* eslint-env jest */
'use strict';

const { defaultsDeep } = require('lodash');
const { generateUrlList, makeBehaviourItem } = require('../cloudfront');

describe('Cloudfront Helpers', () => {
    const testRoutes = {
        sections: {
            purple: {
                path: '/purple',
                pages: {
                    monkey: {
                        path: '/monkey/dishwasher',
                        live: true,
                        isPostable: true,
                        aliases: ['/green/orangutan/fridge'],
                        queryStrings: ['foo', 'bar']
                    }
                }
            }
        },
        archivedRoutes: [
            {
                path: '/some/archived/path/*',
                live: true
            }
        ],
        otherUrls: [
            {
                path: '/unicorns',
                isPostable: true,
                live: true
            },
            {
                path: '/draft',
                live: false
            }
        ]
    };

    describe('#generateUrlList', () => {
        it('should filter out non-custom routes', done => {
            const urls = generateUrlList(testRoutes);
            expect(urls.length).toBe(2);
            done();
        });

        it('should generate the correct section/page path', done => {
            const urls = generateUrlList(testRoutes);
            expect(urls.filter(r => r.path === '/purple/monkey/dishwasher').length).toBe(1);
            done();
        });

        it('should generate welsh versions of canonical routes', done => {
            const urls = generateUrlList(testRoutes);
            expect(urls.filter(r => r.path === '/welsh/purple/monkey/dishwasher').length).toBe(1);
            done();
        });

        it('should store properties against routes', done => {
            const urls = generateUrlList(testRoutes);
            expect(urls.filter(r => r.path === '/purple/monkey/dishwasher')[0].isPostable).toBe(true);
            done();
        });
    });

    describe('#makeBehaviourItem', () => {
        function withDefaults(behaviourConfig) {
            return defaultsDeep(behaviourConfig, {
                MinTTL: 0,
                MaxTTL: 31536000,
                DefaultTTL: 86400,
                FieldLevelEncryptionId: '',
                SmoothStreaming: false,
                Compress: true,
                AllowedMethods: {
                    Items: ['HEAD', 'GET'],
                    Quantity: 2,
                    CachedMethods: {
                        Items: ['HEAD', 'GET'],
                        Quantity: 2
                    }
                },
                ForwardedValues: {
                    QueryStringCacheKeys: {
                        Items: [],
                        Quantity: 0
                    },
                    QueryString: false
                },
                TrustedSigners: {
                    Enabled: false,
                    Items: [],
                    Quantity: 0
                },
                LambdaFunctionAssociations: {
                    Items: [],
                    Quantity: 0
                }
            });
        }

        it('should return cloudfront behaviour for route', () => {
            const behaviour = makeBehaviourItem({
                originId: 'BLF_EXAMPLE',
                pathPattern: '/',
                isPostable: false,
                cookiesInUse: ['example']
            });

            expect(behaviour).toEqual(
                withDefaults({
                    TargetOriginId: 'BLF_EXAMPLE',
                    ViewerProtocolPolicy: 'redirect-to-https',
                    PathPattern: '/',
                    ForwardedValues: {
                        Headers: {
                            Items: ['Accept', 'Host'],
                            Quantity: 2
                        },
                        Cookies: {
                            Forward: 'whitelist',
                            WhitelistedNames: {
                                Items: ['example'],
                                Quantity: 1
                            }
                        },
                        QueryString: true,
                        QueryStringCacheKeys: {
                            Items: ['draft', 'version', 'enable-feature', 'disable-feature'],
                            Quantity: 4
                        }
                    }
                })
            );
        });

        it('should allow all cookies and querystrings', () => {
            const behaviour = makeBehaviourItem({
                originId: 'BLF_LEGACY_EXAMPLE',
                pathPattern: '/~/*',
                isPostable: true,
                allowAllQueryStrings: true,
                allowAllCookies: true,
                protocol: 'allow-all',
                headersToKeep: ['*']
            });

            expect(behaviour).toEqual(
                withDefaults({
                    TargetOriginId: 'BLF_LEGACY_EXAMPLE',
                    ViewerProtocolPolicy: 'allow-all',
                    PathPattern: '/~/*',
                    AllowedMethods: {
                        Items: ['HEAD', 'DELETE', 'POST', 'GET', 'OPTIONS', 'PUT', 'PATCH'],
                        Quantity: 7
                    },
                    ForwardedValues: {
                        Headers: {
                            Items: ['*'],
                            Quantity: 1
                        },
                        Cookies: {
                            Forward: 'all'
                        },
                        QueryStringCacheKeys: {
                            Items: [],
                            Quantity: 0
                        },
                        QueryString: true
                    }
                })
            );
        });
    });
});
