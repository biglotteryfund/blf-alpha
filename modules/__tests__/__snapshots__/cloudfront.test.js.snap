// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "CacheBehaviors": Object {
    "Items": Array [
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "DELETE",
            "POST",
            "GET",
            "OPTIONS",
            "PUT",
            "PATCH",
          ],
          "Quantity": 7,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/api/*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "DELETE",
            "POST",
            "GET",
            "OPTIONS",
            "PUT",
            "PATCH",
          ],
          "Quantity": 7,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "none",
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/funding/grants*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "location",
              "amount",
              "min",
              "max",
            ],
            "Quantity": 7,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/funding/programmes",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "location",
            ],
            "Quantity": 4,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/funding/programmes/all",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "page",
              "tag",
              "author",
              "category",
              "region",
            ],
            "Quantity": 8,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/news/*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/search",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "DELETE",
            "POST",
            "GET",
            "OPTIONS",
            "PUT",
            "PATCH",
          ],
          "Quantity": 7,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "redirectUrl",
              "s",
              "token",
            ],
            "Quantity": 6,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/user/*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "DELETE",
            "POST",
            "GET",
            "OPTIONS",
            "PUT",
            "PATCH",
          ],
          "Quantity": 7,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "none",
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/welsh/funding/grants*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "location",
              "amount",
              "min",
              "max",
            ],
            "Quantity": 7,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/welsh/funding/programmes",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "location",
            ],
            "Quantity": 4,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/welsh/funding/programmes/all",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
              "page",
              "tag",
              "author",
              "category",
              "region",
            ],
            "Quantity": 8,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/welsh/news/*",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "whitelist",
            "WhitelistedNames": Object {
              "Items": Array [
                "blf-alpha-session",
              ],
              "Quantity": 1,
            },
          },
          "Headers": Object {
            "Items": Array [
              "Accept",
              "Host",
            ],
            "Quantity": 2,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [],
            "Quantity": 0,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "/welsh/search",
        "SmoothStreaming": false,
        "TargetOriginId": "ELB_LIVE",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "none",
          },
          "Headers": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
            ],
            "Quantity": 3,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "assets/*",
        "SmoothStreaming": false,
        "TargetOriginId": "S3-blf-assets",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
      Object {
        "AllowedMethods": Object {
          "CachedMethods": Object {
            "Items": Array [
              "HEAD",
              "GET",
            ],
            "Quantity": 2,
          },
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Compress": true,
        "DefaultTTL": 86400,
        "FieldLevelEncryptionId": "",
        "ForwardedValues": Object {
          "Cookies": Object {
            "Forward": "none",
          },
          "Headers": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "QueryString": true,
          "QueryStringCacheKeys": Object {
            "Items": Array [
              "draft",
              "version",
              "social",
            ],
            "Quantity": 3,
          },
        },
        "LambdaFunctionAssociations": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "MaxTTL": 31536000,
        "MinTTL": 0,
        "PathPattern": "media/*",
        "SmoothStreaming": false,
        "TargetOriginId": "S3-blf-assets",
        "TrustedSigners": Object {
          "Enabled": false,
          "Items": Array [],
          "Quantity": 0,
        },
        "ViewerProtocolPolicy": "redirect-to-https",
      },
    ],
    "Quantity": 14,
  },
  "DefaultCacheBehavior": Object {
    "AllowedMethods": Object {
      "CachedMethods": Object {
        "Items": Array [
          "HEAD",
          "GET",
        ],
        "Quantity": 2,
      },
      "Items": Array [
        "HEAD",
        "DELETE",
        "POST",
        "GET",
        "OPTIONS",
        "PUT",
        "PATCH",
      ],
      "Quantity": 7,
    },
    "Compress": true,
    "DefaultTTL": 86400,
    "FieldLevelEncryptionId": "",
    "ForwardedValues": Object {
      "Cookies": Object {
        "Forward": "whitelist",
        "WhitelistedNames": Object {
          "Items": Array [
            "blf-alpha-session",
          ],
          "Quantity": 1,
        },
      },
      "Headers": Object {
        "Items": Array [
          "Accept",
          "Host",
        ],
        "Quantity": 2,
      },
      "QueryString": true,
      "QueryStringCacheKeys": Object {
        "Items": Array [
          "draft",
          "version",
          "social",
        ],
        "Quantity": 3,
      },
    },
    "LambdaFunctionAssociations": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "MaxTTL": 31536000,
    "MinTTL": 0,
    "SmoothStreaming": false,
    "TargetOriginId": "ELB_LIVE",
    "TrustedSigners": Object {
      "Enabled": false,
      "Items": Array [],
      "Quantity": 0,
    },
    "ViewerProtocolPolicy": "redirect-to-https",
  },
}
`;

exports[`#makeBehaviourItem should allow all querystrings 1`] = `
Object {
  "AllowedMethods": Object {
    "CachedMethods": Object {
      "Items": Array [
        "HEAD",
        "GET",
      ],
      "Quantity": 2,
    },
    "Items": Array [
      "HEAD",
      "DELETE",
      "POST",
      "GET",
      "OPTIONS",
      "PUT",
      "PATCH",
    ],
    "Quantity": 7,
  },
  "Compress": true,
  "DefaultTTL": 86400,
  "FieldLevelEncryptionId": "",
  "ForwardedValues": Object {
    "Cookies": Object {
      "Forward": "whitelist",
      "WhitelistedNames": Object {
        "Items": Array [
          "example",
        ],
        "Quantity": 1,
      },
    },
    "Headers": Object {
      "Items": Array [
        "*",
      ],
      "Quantity": 1,
    },
    "QueryString": true,
    "QueryStringCacheKeys": Object {
      "Items": Array [],
      "Quantity": 0,
    },
  },
  "LambdaFunctionAssociations": Object {
    "Items": Array [],
    "Quantity": 0,
  },
  "MaxTTL": 31536000,
  "MinTTL": 0,
  "PathPattern": "/example",
  "SmoothStreaming": false,
  "TargetOriginId": "BLF_EXAMPLE",
  "TrustedSigners": Object {
    "Enabled": false,
    "Items": Array [],
    "Quantity": 0,
  },
  "ViewerProtocolPolicy": "redirect-to-https",
}
`;

exports[`#makeBehaviourItem should return cloudfront behaviour for route 1`] = `
Object {
  "AllowedMethods": Object {
    "CachedMethods": Object {
      "Items": Array [
        "HEAD",
        "GET",
      ],
      "Quantity": 2,
    },
    "Items": Array [
      "HEAD",
      "DELETE",
      "POST",
      "GET",
      "OPTIONS",
      "PUT",
      "PATCH",
    ],
    "Quantity": 7,
  },
  "Compress": true,
  "DefaultTTL": 86400,
  "FieldLevelEncryptionId": "",
  "ForwardedValues": Object {
    "Cookies": Object {
      "Forward": "whitelist",
      "WhitelistedNames": Object {
        "Items": Array [
          "example",
        ],
        "Quantity": 1,
      },
    },
    "Headers": Object {
      "Items": Array [
        "Accept",
        "Host",
      ],
      "Quantity": 2,
    },
    "QueryString": true,
    "QueryStringCacheKeys": Object {
      "Items": Array [
        "draft",
        "version",
        "social",
        "q",
      ],
      "Quantity": 4,
    },
  },
  "LambdaFunctionAssociations": Object {
    "Items": Array [],
    "Quantity": 0,
  },
  "MaxTTL": 31536000,
  "MinTTL": 0,
  "PathPattern": "/",
  "SmoothStreaming": false,
  "TargetOriginId": "BLF_EXAMPLE",
  "TrustedSigners": Object {
    "Enabled": false,
    "Items": Array [],
    "Quantity": 0,
  },
  "ViewerProtocolPolicy": "redirect-to-https",
}
`;
